pkg/auth/oauth/external/github/github.go

```diff
@@ -32,16 +42,41 @@ type githubUser struct {
 	Name  string
 }
 
-func NewProvider(providerName, clientID, clientSecret string) external.Provider {
-	return provider{providerName, clientID, clientSecret}
+// https://developer.github.com/v3/orgs/#response
+type githubOrg struct {
+	ID    uint64
+	Login string
+}
+
+func NewProvider(providerName, clientID, clientSecret string, organizations []string) external.Provider {
+	allowedOrganizations := sets.NewString()
+	for _, org := range organizations {
+		if len(org) > 0 {
+			allowedOrganizations.Insert(strings.ToLower(org))
+		}
+	}
+
+	return &provider{
+		providerName:         providerName,
+		clientID:             clientID,
+		clientSecret:         clientSecret,
+		allowedOrganizations: allowedOrganizations,
+	}
 }
 
-func (p provider) GetTransport() (http.RoundTripper, error) {
+func (p *provider) GetTransport() (http.RoundTripper, error) {
 	return nil, nil
 }
 
 // NewConfig implements external/interfaces/Provider.NewConfig
-func (p provider) NewConfig() (*osincli.ClientConfig, error) {
+func (p *provider) NewConfig() (*osincli.ClientConfig, error) {
+	scopes := []string{}
+	scopes = append(scopes, githubOAuthScope)
+	// if we're limiting to specific organizations, we also need to read their org membership
+	if len(p.allowedOrganizations) > 0 {
+		scopes = append(scopes, githubOrgScope)
```

Comment:
Remind me.  When this changes, users are re-prompted to accept the new scope, right?